notebooks/model-analysis-boston-housing-regression.ipynb

```diff
@@ -0,0 +1,128 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shap\n",
+    "import sklearn\n",
+    "import pandas as pd\n",
+    "\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "\n",
+    "from raiwidgets import ModelAnalysisDashboard\n",
+    "from responsibleai import ModelAnalysis"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use boston housing dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = sklearn.datasets.load_boston()\n",
+    "target_feature = 'y'\n",
+    "continuous_features = data.feature_names\n",
+    "data_df = pd.DataFrame(data.data, columns=data.feature_names)\n",
+    "\n",
+    "X_train, X_test, y_train, y_test = train_test_split(data_df, data.target, test_size=0.2, random_state=7)\n",
+    "\n",
+    "train_data = X_train.copy()\n",
+    "test_data = X_test.copy()\n",
+    "train_data[target_feature] = y_train\n",
+    "test_data[target_feature] = y_test\n",
+    "data.feature_names"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train a model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = RandomForestRegressor()\n",
+    "model.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compute Responsible AI model and dataset insights"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_analysis = ModelAnalysis(model, train_data, test_data, target_feature, 'regression',\n",
+    "                               categorical_features=[])\n",
+    "\n",
+    "# Queue Responsible AI insights\n",
+    "model_analysis.explainer.add()\n",
+    "model_analysis.counterfactual.add(10, desired_range=[10, 20])\n",
+    "model_analysis.error_analysis.add()\n",
+    "model_analysis.causal.add(treatment_features=['ZN'])\n",
```

Comment:
Right now we have a bug for causal that happens on single treatment feature causal.
Do you mind add one more feature here for time being at least?

Reply:
Done